### PR TITLE
Initial draft of offscreen rendering RFC

### PR DESCRIPTION
**Pull request by Pessimistress**

### CODE REVIEW

File: dev-docs/RFCs/v6.0/offscreen-render-rfc.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+WebGL rendering performance can greatly affect UI responsiveness. The experimental [OffscreenCanvas](https://developer.mozilla.org/en-US/docs/Web/API/OffscreenCanvas) API in Chrome and Firefox allows us to move WebGL rendering to a worker thread, and free up the main thread to handle DOM interactions. It is highly desirable for heavy front-end applications.
+
+
+## State of the API
+
+OffscreenCanvas is in active development (behind experimental flags) in both FF and [Chrome](https://www.chromestatus.com/feature/5424182347169792). In Chrome, go to `chrome://flags` and enable "Experimental web platform features."
+
+The API still seems limited/unstable:
+- FF only support OffscreenCanvas with WebGL1 context.
+- Chrome's render buffer does not update in some test scenarios.
```
Comment (ibgreen) [inline dev-docs/RFCs/v6.0/offscreen-render-rfc.md]: Mention the perf issue you found yesterday?

Add Chrome and FF versions and dates as the situation is evolving?

File: dev-docs/RFCs/v6.0/offscreen-render-rfc.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+- `stop()` - equivalent to `animationLoop.stop()`
+
+
+### Examples
+
+```
+/// animation.js
+import {GL, Model, Framebuffer, Matrix4} from 'luma.gl';
+
+export default {
```
Comment (ibgreen) [inline dev-docs/RFCs/v6.0/offscreen-render-rfc.md]: Nit: Would be nice with an interface that avoided the need for such export tricks.

An `AnimationLoop` wrapper class that with an option that resulted in one of these being picked. Understand there is the worker issue...

File: dev-docs/RFCs/v6.0/offscreen-render-rfc.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+```
+
+
+## Proposal: Interactivity
+
+`AnimationLoop` does not directly handle user input (mousemove, click, etc.). This is usually not an issue as the user can update their own application state from event callbacks and then access that state from within `onRender`. In the offscreen rendering case, this is no longer possible as `onRender` is on a different thread from where the event callbacks are executed.
+
+The proposed solution is to expose a new API on both `AnimationLoop` and `OffscreenAnimationLoop` for applications to send custom data to the renderer:
+
+- Add `setUserData` method to `AnimationLoop`. It adds a field `userData` to `animationLoop._callbackData` which can be accessed in the `onRender` callback.
```
Comment (ibgreen) [inline dev-docs/RFCs/v6.0/offscreen-render-rfc.md]: Maybe an interface for forwarding some kind of serialized `mjolnir.js` events?

File: dev-docs/RFCs/v6.0/offscreen-render-rfc.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+- Add `setUserData` method to `OffscreenAnimationLoop`. It posts the user data to the worker which then calls `animationLoop.setUserData`.
+
+
+## Challenges
+
+The main challenge in this effort is that DOM properties are accessed in various places of the code base (e.g. webgl-utils, AnimationLoop, hover/click handling, viewport controls). Certain globals such as `window` `document` and `HTMLCanvasElement` are not available to the worker thread. We should either improve the compatibility of existing code, or invest in separating DOM-dependent code from the core.
```
Comment (ibgreen) [inline dev-docs/RFCs/v6.0/offscreen-render-rfc.md]: This has been resolved, right? Maybe change the tense of the sentence to "maintaining" or similar.

File: dev-docs/RFCs/v6.0/offscreen-render-rfc.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+## Proposal: OffscreenAnimationLoop
+
+Add an `OffscreenAnimationLoop` that mirrors the `AnimationLoop` class but delegate rendering to a worker.
+
+Static methods:
+- `createWorker(opts)` - returns script that can be used to spin up a new Worker. `opts` are the options used to construct an `AnimationLoop` instance on the worker thread.
+
+Constructor:
+- `new OffscreenAnimationLoop(opts)` - `opts` are the same options used to construct an `AnimationLoop`, with additional fields:
```
Comment (ibgreen) [inline dev-docs/RFCs/v6.0/offscreen-render-rfc.md]: Nit: We can postpone this till audit but I think we were at some point considering a name like `OffThreadAnimationLoop` , `WorkerThreadAnimationLoop`, `ThreadedAnimationLoop` or `WorkerAnimationLoop`?

File: dev-docs/RFCs/v6.0/offscreen-render-rfc.md (hunk 3 above)
Comment (Pessimistress) [reply dev-docs/RFCs/v6.0/offscreen-render-rfc.md]: That sounds arbitrary as `mjolnir.js` is not a luma.gl dependency.

File: dev-docs/RFCs/v6.0/offscreen-render-rfc.md (hunk 4 above)
Comment (Pessimistress) [reply dev-docs/RFCs/v6.0/offscreen-render-rfc.md]: Not entirely. We still have these scattered in the code base, e.g. picking.

**Pull request merged**